support/src/lib.rs

```diff
@@ -17,5 +17,17 @@
 // If you feel like getting in touch with us, you can do so at info@botlabs.org
 #![cfg_attr(not(feature = "std"), no_std)]
 
+use deposit::Deposit;
+use frame_support::traits::{Currency, ReservableCurrency};
+use sp_runtime::traits::Zero;
+
 pub mod deposit;
 pub mod traits;
+
+pub fn free_deposit<A, C>(deposit: &Deposit<A, C::Balance>)
```

Comment:
C: Use this function in other pallets as well. But I suppose this will be handled in a separate PR.